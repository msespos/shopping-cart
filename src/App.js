import NavBar from "./NavBar";
import VAAimg from "./imgs/VintageAutechreAlbums.png"
import "./Styles/App.css"

const App = () => {
  return (
    <div>
      <NavBar />
      <div className="image-and-blurb">
        <img className="image" src={VAAimg} alt="Vintage Autechre Albums" width={300} />
        <div className="blurb">
          Autechre have been true musical pioneers for over thirty years.
          Buy digital copies of their first 12 albums here.<br></br><br></br>
          If you've never heard Autechre before, strap in - you're in for an
          experience like no other!
        </div>
      </div>
    </div>
  );
}

export default App;
