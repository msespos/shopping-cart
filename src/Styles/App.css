.image-and-blurb {
  display: flex;
  justify-content: space-between;
  margin: 30px;
  gap: 50px;
}

@media (max-width: 1250px) {
  .image-and-blurb {
    flex-direction: column;
  }
}

.image {
  width: 500px;
  height: 400px;
}

@media (max-width: 1250px) {
  .image {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 300px;
    height: 240px;
  }
}

@media (max-width: 1250px) {
  .blurb {
    margin-top: 0;
    font-size: 25px;
  }
}
