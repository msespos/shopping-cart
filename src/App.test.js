import { render, screen } from "@testing-library/react";
import { HashRouter } from "react-router-dom";
import App from "./App";
import '@testing-library/jest-dom';

describe("App component", () => {
  it("renders the component", () => {
    const component = render(<App />, {wrapper: HashRouter});
    expect(component).toMatchSnapshot();
  });

  it("has the correct source for the body image", () => {
    render(<App />, {wrapper: HashRouter});
    const bodyImage = screen.getByAltText(/Vintage Autechre Albums/i);
    const imageSource = "VintageAutechreAlbums.png";
    expect(bodyImage).toHaveAttribute("src", imageSource);
  });
});
