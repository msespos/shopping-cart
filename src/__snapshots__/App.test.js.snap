// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            class="note"
          >
            <em>
              Note: this is not an actual store. It's simply a site created for learning purposes. I'm not actually selling anything, least of all Autechre's music!
              <br />
              But you should definitely go listen to Autechre right now!
            </em>
          </div>
          <div
            class="navbar"
          >
            <a
              href="#/"
            >
              <img
                alt="VAA logo"
                src="logo.png"
                width="50"
              />
            </a>
            <a
              href="#/"
            >
              Home
            </a>
            <a
              href="#/shop"
            >
              Shop
            </a>
          </div>
        </div>
        <div
          class="image-and-blurb"
        >
          <img
            alt="Vintage Autechre Albums"
            class="image"
            src="VintageAutechreAlbums.png"
            width="300"
          />
          <div
            class="blurb"
          >
            Autechre have been true musical pioneers for over thirty years. Buy digital copies of their first 12 albums here.
            <br />
            <br />
            If you've never heard Autechre before, strap in - you're in for an experience like no other!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          class="note"
        >
          <em>
            Note: this is not an actual store. It's simply a site created for learning purposes. I'm not actually selling anything, least of all Autechre's music!
            <br />
            But you should definitely go listen to Autechre right now!
          </em>
        </div>
        <div
          class="navbar"
        >
          <a
            href="#/"
          >
            <img
              alt="VAA logo"
              src="logo.png"
              width="50"
            />
          </a>
          <a
            href="#/"
          >
            Home
          </a>
          <a
            href="#/shop"
          >
            Shop
          </a>
        </div>
      </div>
      <div
        class="image-and-blurb"
      >
        <img
          alt="Vintage Autechre Albums"
          class="image"
          src="VintageAutechreAlbums.png"
          width="300"
        />
        <div
          class="blurb"
        >
          Autechre have been true musical pioneers for over thirty years. Buy digital copies of their first 12 albums here.
          <br />
          <br />
          If you've never heard Autechre before, strap in - you're in for an experience like no other!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
